src/Algebra/Graph/HigherKinded/Class.hs

```diff
@@ -436,6 +359,23 @@ star :: Graph g => a -> [a] -> g a
 star x [] = vertex x
 star x ys = connect (vertex x) (vertices ys)
 
+-- | The /stars/ formed by overlaying a list of 'star's. An inverse of
+-- 'adjacencyList'.
+-- Complexity: /O(L)/ time, memory and size, where /L/ is the total size of the
+-- input.
+--
+-- @
+-- stars []                      == 'empty'
+-- stars [(x, [])]               == 'vertex' x
+-- stars [(x, [y])]              == 'edge' x y
+-- stars [(x, ys)]               == 'star' x ys
+-- stars                         == 'overlays' . map (uncurry 'star')
+-- stars . 'adjacencyList'         == id
+-- 'overlay' (stars xs) (stars ys) == stars (xs ++ ys)
+-- @
+stars :: Graph g => [(a, [a])] -> g a
+stars = overlays . map (uncurry star)
```

Comment:
Does it also make sense to switch to your better `overlays` implementation based on `foldr1Safe`?

Reply:
I don't know, I think yes ! I will do some benchs tomorrow

Reply:
Mmhh, I spoke too quickly, the `concatg` solution seems to not work, at least not out-of-the-box...
It will certainly require further work, maybe this is linked with the same problem in `Algebra.Graph.Fold`